src/hyp_layout.rs

```diff
@@ -75,6 +86,17 @@ const_assert!(PageSize::Size4k.is_aligned(UMODE_INPUT_START));
 /// Size of the U-mode Input Region.
 pub const UMODE_INPUT_SIZE: u64 = 4 * 1024;
 const_assert!(PageSize::Size4k.is_aligned(UMODE_INPUT_SIZE));
+/// End of the U-mode Input Region.
+const UMODE_INPUT_END: u64 = UMODE_INPUT_START + UMODE_INPUT_SIZE;
+/// Last U-mode address.
+const UMODE_END: u64 = UMODE_INPUT_END;
+
+/// The lowest possible kernel stack address.
+pub const HYP_STACKBTM_MIN: u64 = UMODE_END + 4 * 1024 * 1024;
```

Comment:
is the difference between "bottom min" and `top - stack size" because cpu0 has a different stack size?

Reply:
Yes, or rather, was. I want to fix that in the next PR, where I force the stack to be the same on all CPUs (to ease calculation on trap handler for stack bottom). This was just a check on the bootstrap CPU stack (that is controlled at the moment by the linker script)

Reply:
All CPUs have the same size in the new iteration.